dcicutils/misc_utils.py

```diff
@@ -2310,3 +2330,43 @@ def parse_in_radix(text: str, *, radix: int):
     except Exception:
         pass
     raise ValueError(f"Unable to parse: {text!r}")
+
+
+def pad_to(target_size: int, data: list, *, padding=None):
+    actual_size = len(data)
+    if actual_size < target_size:
+        data = data + [padding] * (target_size - actual_size)
+    return data
+
+
+class JsonLinesReader:
+
+    def __init__(self, fp, padded=False, padding=None):
+        self.fp = fp
```

Comment:
Consider renaming; I believe this refers to the schema file path.

Reply:
`fp` is the conventional name in Python for what most languages call a `stream`. I might be willing to call it that, too. I think the name `fp` is stupid. An `fp` is an open "file pointer".

Reply:
I have (hopefully) addressed this by a doc string, which begins

> Given an fp (the conventional name for a "file pointer", the thing a call to io.open returns, this creates an object that can be used to iterate across the lines in the JSON lines file that the fp is reading from.